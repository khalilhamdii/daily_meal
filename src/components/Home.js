import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import homeBg from '../assets/home_bg.jpg';

const HomeDiv = styled.div`
  background: linear-gradient(rgba(0, 0, 0, 0.8), rgba(0, 0, 0, 0.8)),
    url(${homeBg}) no-repeat;
  background-size: auto, cover;
`;

const Home = () => (
  <HomeDiv className="vw-100 vh-100 primary-color d-flex align-items-center justify-content-center text-white">
    <div className="jumbotron jumbotron-fluid bg-transparent">
      <div className="container secondary-color">
        <h1 className="display-4">Daily meal</h1>
        <p className="lead">
          A curated list of recipes for the best homemade meal and delicacies.
        </p>
        <hr className="my-4" />
        <Link
          to="/categories"
          className="btn btn-lg text-white"
          role="button"
        >
          View Categories
        </Link>
      </div>
    </div>
  </HomeDiv>
);

export default Home;
