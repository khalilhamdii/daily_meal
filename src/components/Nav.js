import React from 'react';
import '../assets/nav.css';

function Nav() {
  return (
    <nav className="navbar navbar-light navbar-expand-md navigation-clean-button">
      <div className="container">
        <a className="navbar-brand" href="/">
          Daily meals
        </a>
        <button
          data-toggle="collapse"
          className="navbar-toggler"
          type="button"
          data-target="#navcol-1"
        >
          <span className="sr-only">Toggle navigation</span>
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navcol-1">
          <ul className="nav navbar-nav mr-auto">
            <li className="nav-item" />
          </ul>
          <i
            className="fa fa-user-circle fa-2x ml-4"
            style={{ color: 'white' }}
          />
        </div>
      </div>
    </nav>
  );
}

export default Nav;
