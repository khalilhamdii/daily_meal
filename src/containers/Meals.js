/* eslint-disable react/prop-types */
/* eslint-disable arrow-parens */
import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import styled from 'styled-components';
import Meal from '../components/Meal';
import mealsBg from '../assets/meals-bg.jpg';
import { filtredMealsToProps } from '../helpers/index';
import { changeCategoryFilter, changeAreaFilter } from '../actions/index';
import MealsFilter from '../components/MealsFilter';
import Nav from '../components/Nav';

const MealsDiv = styled.div`
  background: linear-gradient(rgba(0, 0, 0, 0.8), rgba(0, 0, 0, 0.8)),
    url(${mealsBg}) no-repeat;
  background-size: auto, cover;
  min-height: 100vh;
`;
const Meals = (props) => {
  const { meals, filter } = props;
  console.log(meals);
  const handleCatFilterChange = (filter) => {
    props.changeCategoryFilter(filter);
  };
  const handleAreaFilterChange = (filter) => {
    props.changeAreaFilter(filter);
  };
  return (
    <>
      <Nav />
      <MealsDiv className="py-5">
        <main className="container">
          <MealsFilter
            filter={filter}
            handleCatFilterChange={handleCatFilterChange}
            handleAreaFilterChange={handleAreaFilterChange}
          />
          <Link to="/" className="btn back-link my-5">
            <i className="fa fa-reply mr-2" />
            Back to Home
          </Link>
          <div className="row">
            {meals.map((meal) => (
              <Meal key={meal.idMeal} meal={meal} />
            ))}
          </div>
        </main>
      </MealsDiv>
    </>
  );
};

export default connect(filtredMealsToProps, {
  changeCategoryFilter,
  changeAreaFilter,
})(Meals);
