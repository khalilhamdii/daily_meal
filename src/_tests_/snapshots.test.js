import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Nav from '../components/Nav';
import Home from '../components/Home';
import Meals from '../containers/Meals';

describe('Snapshots', () => {
  it('renders Home correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders Meals correctly', () => {
    const tree = renderer.create(<Meals />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
