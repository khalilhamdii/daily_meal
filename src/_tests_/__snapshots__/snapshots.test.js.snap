// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots renders Home correctly 1`] = `
<div
  className="sc-bdfBwQ IqGYf vw-100 vh-100 primary-color d-flex align-items-center justify-content-center text-white"
>
  <div
    className="jumbotron jumbotron-fluid bg-transparent"
  >
    <div
      className="container secondary-color"
    >
      <h1
        className="display-4"
      >
        Daily meal
      </h1>
      <p
        className="lead"
      >
        A curated list of recipes for the best homemade meal and delicacies.
      </p>
      <hr
        className="my-4"
      />
      <a
        className="view-meals"
        href="/meals"
      >
        <h4>
          View meals
        </h4>
      </a>
    </div>
  </div>
</div>
`;
